### DISCUSSION TITLE
How to parse Type1C font file extract from a PDF file?

### DISCUSSION THREAD

**Discussion by prRZ5F4LXZ** [fonttools/fonttools#3483] [General]
out.font is extracted from out.pdf using the following command.

cpdf -extract-font 1,/R7 out.pdf -o out.font

Is there a good way to convert out.font to XML format?

[out.pdf](https://github.com/fonttools/fonttools/files/14985240/out.pdf)

**Comment by ftCLI** [comment]
```
makeotf -f out.font -o out.otf
ttx out.otf
```

**Reply by prRZ5F4LXZ** [reply]
I see these tables in out.otf. Are they available in out.font? Or some of them are added by makeotf?

GlyphOrder
head
hhea
maxp
OS_2
name
cmap
post
CFF
hmtx

Do the warnings below cause any info missing in the out.otf file?

$ makeotf -f out.font -o out.otf
tx: --- /private/tmp/out.font
tx: (cfr) /UnderlineThickness invalid (ignored)
tx: --- /private/tmp/out.font
tx: (cfr) /UnderlineThickness invalid (ignored)
makeotf [Warning] Could not find default features file. Font will be built without any layout features.
makeotf [Warning] Could not find FontMenuNameDB file. Font will be built with menu names derived from PostScript name.
makeotfexe processing font <Courier>
WARNING: FontMenuNameDB file was not specified or not found. [Courier]
WARNING: Negative internal leading: win.ascent + win.descent < unitsPerEm
Built development mode font '/private/tmp/out.otf'.
